docs/user-guide/expressions/numpy.md

```diff
@@ -13,10 +13,26 @@ This means that if a function is not provided by Polars, we can use NumPy and we
 --8<-- "python/user-guide/expressions/numpy-example.py"
 ```
 
+## Numba
+
+[Numba](https://numba.pydata.org/) is an open source JIT compiler that allows you to create your own ufuncs entirely within python. The key is to use the [@guvectorize](https://numba.readthedocs.io/en/stable/user/vectorize.html#the-guvectorize-decorator) decorator. One popular use case is conditional cumulative functions. For example, suppose you want to take a cumulative sum but have it reset whenever it gets to a threshold.
+
+### Example
+
+{{code_block('user-guide/expressions/numpy-example',api_functions=['DataFrame'])}}
+
+```python exec="on" result="text" session="user-guide/numpy"
+--8<-- "python/user-guide/expressions/numba-example.py"
+```
+
 ### Interoperability
 
 Polars `Series` have support for NumPy universal functions (ufuncs). Element-wise functions such as `np.exp()`, `np.cos()`, `np.div()`, etc. all work with almost zero overhead.
 
 However, as a Polars-specific remark: missing values are a separate bitmask and are not visible by NumPy. This can lead to a window function or a `np.convolve()` giving flawed or incomplete results.
 
 Convert a Polars `Series` to a NumPy array with the `.to_numpy()` method. Missing values will be replaced by `np.nan` during the conversion.
+
+### Note on Performance
+
+The speed of ufuncs comes from being vectorized, and compiled. That said, there's no inherent benefit in using ufuncs just to avoid the use of `map_batches`. As mentioned above, ufuncs use a hook which gives polars the opportunity to run its own code before the ufunc is executed. In that way polars is still executing the ufunc with `map_batches`.
```

Comment:
true, but `map_batches` has `is_elementwise=False` by default, and so will do the expected thing in group-by / over

maybe, rather than avoiding `map_batches`, `map_batches` should be the primary way the ufuncs are taught?

Reply:
@MarcoGorelli 

It turns out that all the numpy and scipy ufuncs are element wise in the sense that they aren't aggregations. Where this becomes important is if someone wants to do a sum and then the np.expm1 function. The ufuncarray hook will do is_elementwise=true. If it's anumba ufunc then it won't.

Whether people should be taught to not use the hook is more philosophical imo. I suppose there's slightly less parsing in that case so would be technically more performant but by using it, it's a nice syntax, imo.